### PR TITLE
Add Log escalation failure if SF bot is offline/disabled

### PR DESCRIPTION
**Pull request by ItzzRitik**
#### Steps to reproduce these changes
1. Make sure that Salesforce Bot is offline, if not log out from the Salesforce bot.
2. Disabling the bot will have a similar effect as it will make the bot offline.
3. That's it, now try to escalate the case from livechat widget.
4. It should fail and log the error **Failed to handover**, as shown below

#### Screenshot of the logpoint

image

### CODE REVIEW

File: lib/Room.ts [hunk 1]
```diff
@@ -84,7 +84,16 @@ export const performHandover = async (modify: IModify, read: IRead, rid: string,
...
+        const offlineMessage: string = await getAppSettingValue(read, AppSetting.DialogflowServiceUnavailableMessage),
+        handoverFailure = {
+            error: offlineMessage,
+            errorMessage: 'Unable to reach Liveagent bot, it may be offline or disabled.',
+            dialogflow_SessionID: rid,
+            visitorDetails: (({ id, token }) => ({ id, token }))(visitor),
+            targetDepartment: livechatTransferData.targetDepartment
+        }
+
+        console.log('Failed to handover', handoverFailure);
```
Comment (ear-dev) [inline lib/Room.ts:96]: @ItzzRitik can you use this from now on when we add new log points please..... JSON.stringify(handoverFailure)

Comment (ItzzRitik) [reply lib/Room.ts:96]: Understood, will take care

Comment (ear-dev) [reply lib/Room.ts]: thanks....  I'll test

Comment (ear-dev) [reply lib/Room.ts]: @ItzzRitik I'm not yet able to get this to work....  maybe tomorrow you can demo for us how this should work.

Comment (ItzzRitik) [reply lib/Room.ts]: @ear-dev I have updated the description, with steps to reproduce, please have a look.

**Comment by ear-dev** [comment]
@ItzzRitik can you be more precise when you say "Make sure that Salesforce Bot is offline, if not log out from the Salesforce bot."?

- Do you mean that the "salesforce-live-agent-integration.bot" user is offline?
- Do you mean that the "Salesforce Live Agent Integration" app is disabled?

Also, in the case where the Dialogflow app is misconfigured to handover to a department that does not exist, will this trigger the failed handover event? 

Can you confirm other scenarios as well please:

- salesforce app has no available agents
- salesforce app is broken/frozen
- salesforce app is misconfigured so that it cannot talk to the salesforce backend properly

I know that there are a lot of edge cases...... 

Can you please create a document that describes every case where your failed to handover log points will be triggered?
I'll make a story about that.

**Comment by ItzzRitik** [comment]
@ear-dev That's a great plan, would be better to keep everything at a place.
I will surely make a document capturing all **Failed to Handover** events.
and I'm aware that this task is being tracked [here](https://github.com/WideChat/Rocket.Chat/issues/631).
Thanks!

**Pull request merged**